src/bio/bio_xstream.c

```diff
@@ -1472,25 +1554,34 @@ bio_xsctxt_alloc(struct bio_xs_context **pctxt, int tgt_id, bool self_polling)
 		}
 	}
 
-	/* Initialize per-xstream blobstore context */
+	/* Initialize main xstream blobstore context */
 	for (st = SMD_DEV_TYPE_DATA; st < SMD_DEV_TYPE_MAX; st++) {
-		/* only data blob now */
-		if (st != SMD_DEV_TYPE_DATA)
+		/* No Data blobstore for sys xstream */
+		if (st == SMD_DEV_TYPE_DATA && tgt_id == DSS_SYS_TGT_ID)
 			continue;
+
+		/* No Meta/WAL blobstore if Metadata on SSD is not configured */
+		if (st != SMD_DEV_TYPE_DATA && !bio_is_meta_on_ssd_configured())
+			break;
+
 		rc = init_xs_blobstore_ctxt(ctxt, tgt_id, st);
 		if (rc)
 			goto out;
 	}
 
+	if (tgt_id == DSS_SYS_TGT_ID) {
+		rc = bio_sys_smd_init();
+		if (rc)
+			goto out;
+	}
+
```

Comment:
I didn't see SMD updated after assign blobstore to sys xstream.

Reply:
You mean we also need write SMD for sys stream mapping? i think we discussed that you mentioned that this will  be not used even we updated SMD for sys xs.